src/components/Articles.jsx

```diff
@@ -4,12 +4,23 @@ import {getArticles} from '../utils/api_funcs'
 import ArticleCard from "./ArticleCard"
 
 const Articles = () =>{
+    const search = window.location.search;
     
     const [articlesList, setArticlesList] = useState([])
+  
     useEffect(()=> {
-       getArticles().then((response)=> {setArticlesList(response.articles)})
+       getArticles().then((response)=> { if(search) {
+        const topicName = search.slice(7)
+        const filteredArticles = response.articles.filter((article) => {
+            return article.topic === topicName
+        })
+        setArticlesList(filteredArticles)
+    } else {
+        setArticlesList(response.articles)}})
```

Comment:
You're best looking into useSearchParams. 

You might have an issue with this later on when trying to do the sort/order. They'll end up getting caught up in this, as the url will look like `website.com?sortBy=`